From da5f1cd23610f6d86adc7f2d30e7feb91beb2ca2 Mon Sep 17 00:00:00 2001
From: Eric Biggers <ebiggers@google.com>
Date: Wed, 12 Aug 2020 16:11:53 -0700
Subject: [PATCH] _____ANNOTATION: kbuild changes_____

Signed-off-by: Eric Biggers <ebiggers@google.com>
Signed-off-by: Lee Jones <joneslee@google.com>
Change-Id: If01e2f09d681f3a36e69a701a46f02bb5ccace52
---
 ANNOTATIONS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ANNOTATIONS b/ANNOTATIONS
index 64d54527a41c..add586f8c40e 100644
--- a/ANNOTATIONS
+++ b/ANNOTATIONS
@@ -1,2 +1,3 @@
 pending upstream
 binder changes
+kbuild changes
-- 
2.34.1

