From 0c16162dac4c9c787e4848bf4e1ca09f4837431e Mon Sep 17 00:00:00 2001
From: Eric Biggers <ebiggers@google.com>
Date: Wed, 12 Aug 2020 14:00:49 -0700
Subject: [PATCH] _____ANNOTATION: sound changes_____

Signed-off-by: Eric Biggers <ebiggers@google.com>
Signed-off-by: Lee Jones <joneslee@google.com>
Change-Id: I2839c2cff2f4bc618219c98009624102b5a1e2f1
---
 ANNOTATIONS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ANNOTATIONS b/ANNOTATIONS
index 72e03a946419..c95dffb5e1ae 100644
--- a/ANNOTATIONS
+++ b/ANNOTATIONS
@@ -12,3 +12,4 @@ driver framework and misc driver changes
 tty changes
 USB changes
 graphics changes
+sound changes
-- 
2.34.1

