From 426ae258b1dbf29538c0e314e1f3339c6d1bc848 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Fri, 30 Apr 2021 10:53:12 +0300
Subject: [PATCH 4/6] rockchip: Use default fastboot_set_reboot_flag
 implementation

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 arch/arm/mach-rockchip/board.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/arch/arm/mach-rockchip/board.c b/arch/arm/mach-rockchip/board.c
index f1f70c81d0..c13eb37c4e 100644
--- a/arch/arm/mach-rockchip/board.c
+++ b/arch/arm/mach-rockchip/board.c
@@ -306,20 +306,6 @@ int board_usb_init(int index, enum usb_init_type init)
 
 #endif /* CONFIG_USB_GADGET */
 
-#if IS_ENABLED(CONFIG_FASTBOOT)
-int fastboot_set_reboot_flag(enum fastboot_reboot_reason reason)
-{
-	if (reason != FASTBOOT_REBOOT_REASON_BOOTLOADER)
-		return -ENOTSUPP;
-
-	printf("Setting reboot to fastboot flag ...\n");
-	/* Set boot mode to fastboot */
-	writel(BOOT_FASTBOOT, CONFIG_ROCKCHIP_BOOT_MODE_REG);
-
-	return 0;
-}
-#endif
-
 #ifdef CONFIG_MISC_INIT_R
 __weak int misc_init_r(void)
 {
-- 
2.37.2

